pandas/io/parsers.py

```diff
@@ -3036,7 +3044,7 @@ def _search_replace_num_columns(self, lines, search, replace):
                     not isinstance(x, str)
                     or search not in x
                     or (self._no_thousands_columns and i in self._no_thousands_columns)
-                    or self.nonnum.search(x.strip())
+                    or not self.nonnum.search(x.strip())
```

Comment:
this flips the meaning of nonnum right?  should rename?

Reply:
Yep is better, done